src/Agent.Worker/WorkerCommandManager.cs

```diff
@@ -133,6 +133,49 @@ public interface IWorkerCommandExtension : IExtension
         void ProcessCommand(IExecutionContext context, Command command);
     }
 
+    public interface IWorkerCommand 
+    {
+        string Name { get; }
+
+        string[] Aliases { get; }
+
+        void Execute(IExecutionContext context, Command command);
+    }
+
+    public abstract class BaseWorkerCommandExtension: AgentService, IWorkerCommandExtension
+    {
+
+        public string CommandArea { get; protected set; }
+
+        public HostTypes SupportedHostTypes { get; protected set; }
+
+        public Type ExtensionType => typeof(IWorkerCommandExtension);
+
+        private Dictionary<string, IWorkerCommand> _commands = new Dictionary<string, IWorkerCommand>(StringComparer.OrdinalIgnoreCase);
+
+        protected void InstallWorkerCommand(IWorkerCommand commandExecutor)
+        {
+            // TODO: check for already defined
+            _commands[commandExecutor.Name] = commandExecutor;
+            foreach (var alias in commandExecutor.Aliases)
+            {
+                _commands[alias] = commandExecutor;
+            }
+        }
+
+        public void ProcessCommand(IExecutionContext context, Command command)
+        {
+            IWorkerCommand commandExecutor = null;
+            _commands.TryGetValue(command.Event, out commandExecutor);
```

Comment:
in c# you don't have to forward declare the output pointer. 
you can say: 
```csharp
_commands.TryGetValue(command.Event, out var commandExecutor);
if (commandExecutor == null) {...}
```

Reply:
FWIW that's new in C# 7, which was new in 2017 (likely after some of the agent code was written).